src/query/encoding.ts

```diff
@@ -48,3 +48,38 @@ export function isMeasure(encQ: EncodingQuery) {
   return (encQ.type === Type.QUANTITATIVE && !encQ.bin) ||
       (encQ.type === Type.TEMPORAL && !encQ.timeUnit);
 }
+
+/**
+ *  Returns the true scale type of an encoding.
+ *  @returns {ScaleType} If the scale type was not specified, it is inferred from the encoding's Type.
+ *  @returns {undefined} If the scale type was not specified and Type (or TimeUnit if applicable) is an EnumSpec, there is no clear scale type
+ */
+
+export function scaleType(scaleType: ScaleType, timeUnit: TimeUnit | EnumSpec<TimeUnit> | ShortEnumSpec, type: Type | EnumSpec<Type> | ShortEnumSpec) {
```

Comment:
Why the other two parameters can be enum spec but scaleType cannot be enum spec? 

I wonder if you can actually simplify this method to never accept enum spec? 
(Im not sure about that you need to check all reference if you always have them in specific form.)